pkg/component/config/genericcomponent.go

```diff
@@ -47,7 +48,11 @@ func NewDefaultGenericComponentConfiguration() GenericComponentConfiguration {
 		KubeAPIBurst:            30,
 		ControllerStartInterval: metav1.Duration{Duration: 0 * time.Second},
 	}
-	leaderElection := componentbaseconfigv1alpha1.LeaderElectionConfiguration{}
+	leaderElection := componentbaseconfigv1alpha1.LeaderElectionConfiguration{
+		// https://github.com/kubernetes/kubernetes/pull/84084
+		// https://github.com/kubernetes/kubernetes/pull/106852
+		ResourceLock: resourcelock.EndpointsLeasesResourceLock,
```

Comment:
change the default resource lock to `endpointsleases` as  https://github.com/kubernetes/kubernetes/pull/106852/files#diff-a25ac8933468a654f5d22934e006e00f17a64fd414fb2371540b122d661c4b1eR193